module/x/gravity/keeper/keeper_test.go

```diff
@@ -189,20 +189,23 @@ func TestLastSlashedValsetNonce(t *testing.T) {
 	lastSlashedValsetNonce = k.GetLastSlashedValsetNonce(ctx)
 	assert.Equal(t, lastSlashedValsetNonce, uint64(3))
 
-	// when maxHeight < lastSlashedValsetNonce, len(unslashedValsets) should be zero
-	unslashedValsets = k.GetUnSlashedValsets(ctx, uint64(2))
+	lastSlashedValset := k.GetValset(ctx, lastSlashedValsetNonce)
+
+	// when valset height + signedValsetsWindow > current block height, len(unslashedValsets) should be zero
+	unslashedValsets = k.GetUnSlashedValsets(ctx, uint64(ctx.BlockHeight()))
 	assert.Equal(t, len(unslashedValsets), 0)
 
-	// when maxHeight == lastSlashedValsetNonce, len(unslashedValsets) should be zero
-	unslashedValsets = k.GetUnSlashedValsets(ctx, uint64(3))
+	// when lastSlashedValset height + signedValsetsWindow == BlockHeight, len(unslashedValsets) should be zero
```

Comment:
Justin please check the decisions I made here carefully, I am not 100% certain of them.

I tried to duplicate these situations:
1. Get no unslashed valsets when the window is too large
2. Get no unslashed valsets when the window is exactly the right size
3. Get 2 unslashed valsets when the window ends after the 2nd unslashed valset
4. Get all the unslashed valsets when the window ends after all of them.